meson.build

```diff
@@ -0,0 +1,86 @@
+project(
+	'dunst',
+	'c',
+	version: '1.9.2',
+	license: 'MIT',
+	meson_version: '>=0.60.0',
+	default_options: [
+		'c_std=gnu99',
+		'warning_level=1',
+		'b_ndebug=if-release',
+	],
+)
+
+add_project_arguments([
+	'-pedantic',
+	'-Wno-gnu-zero-variadic-macro-arguments',
+	'-Wno-overlength-strings',
+], language: 'c')
+
+if get_option('buildtype').startswith('debug')
+	add_project_arguments('-DDEBUG_BUILD', language: 'c')
+endif
+
+cc = meson.get_compiler('c')
+
+cairo = dependency('cairo')
+glib = dependency('glib-2.0')
+gio = dependency('gio-2.0')
+gdk_pixbuf = dependency('gdk-pixbuf-2.0')
+pangocairo = dependency('pangocairo')
+x11 = dependency('x11')
+xinerama = dependency('xinerama')
+xext = dependency('xext')
+xrandr = dependency('xrandr', version: '>=1.5')
+xscrnsaver = dependency('xscrnsaver')
+systemd = dependency('systemd', required: get_option('systemd'))
+libnotify = dependency('libnotify', required: get_option('dunstify'))
+realtime = cc.find_library('rt')
+math = cc.find_library('m')
+wayland_client = dependency('wayland-client', required: get_option('wayland'))
+wayland_protos = dependency('wayland-protocols', version: '>=1.32', required: get_option('wayland'))
+wayland_cursor = dependency('wayland-cursor', required: get_option('wayland'))
+
+c_version_arg = '-DVERSION="@0@"'.format(meson.project_version())
+
+subdir('src')
+
+install_data('dunstctl', install_dir: get_option('bindir'))
+install_data('dunstrc', install_dir: get_option('sysconfdir') / 'dunst')
```

Comment:
This reverts move to `/etc/xdg/dunst/` and installs the config as `/etc/dunst/dunstrc`, right?

Maybe meson should just use `get_option('sysconfdir') / 'xdg'` both here and in conf_data unconditionally.

Reply:
Also needs something like
```meson
prefix = get_option('prefix')

add_project_arguments(
    '-DSYSCONFDIR="@0@"'.format(prefix / get_option('sysconfdir') / 'xdg'),
    language: 'c'
)
...
conf_data.set('bindir', prefix / get_option('bindir'))
conf_data.set('sysconfdir', prefix /get_option('sysconfdir'))
```
(get_option('...dir') won't include prefix when used in a string context)

Reply:
Or just put it in `/usr/share/dunst/dunstrc`...

Reply:
On a seperate note, why is `SYSCONFDIR` supposed to be defined? it seems to be used only for a Doxygen workaround.

Reply:
It's used in `src/settings.c:get_xdg_conf_basedirs` to pick the correct default configuration path. If you don't define `SYSCONFDIR`, dunst will look for configuration in `/usr/local/etc/xdg`, which is a bit... BSD'ish.

> It is never mentioned nor added in the Makefile.

See `DEFAULT_CPPFLAGS` in the [config.mk](https://github.com/dunst-project/dunst/blob/master/config.mk)

Reply:
Yeah, /usr/local/etc/xdg is a bit long for my tastes as well, but it's best to leave this the same for now. If you want to change that, it can be discussed in another PR

Reply:
`PREFIX/etc` is not a valid directory in the Linux FHS. I do not understand the need for it to be in `/usr/local/etc` or `/usr/etc`.

Reply:
Right, that's one of the confusing parts of meson. `get_option('prefix') / get_option('sysconfdir')` works quite counter-intuitive, but it's the right thing to use.

Consider:
| meson setup args | `get_option('prefix')` | `get_option('sysconfdir')` | `get_option('prefix') / get_option('sysconfdir')` | `install_dir: get_option('sysconfdir')`
| --- | --- | --- | --- | ---
| default | `/usr/local` | `etc` | `/usr/local/etc` | `/usr/local/etc`
| `--prefix=/usr/local` | `/usr/local` | `etc` | `/usr/local/etc` | `/usr/local/etc`
| `--prefix=/usr` | `/usr` | `/etc` | `/etc` | `/etc`
| `--sysconfdir=/etc` | `/usr/local` | `/etc` | `/etc` | `/etc`

With the current checkout, `man build/docs/dunst.1` will show you that the substituted value was `etc/xdg`, which is not what we want.